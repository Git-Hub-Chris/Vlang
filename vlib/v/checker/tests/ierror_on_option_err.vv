struct Struct {
	error ?string
}

fn result_function() !Struct {
	return error('This is an error')
}

fn main() {
	some_struct := result_function() or {
		Struct{
			error: err
		}
	}
	println(some_struct)
}
