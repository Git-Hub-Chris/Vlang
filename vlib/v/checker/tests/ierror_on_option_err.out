vlib/v/checker/tests/ierror_on_option_err.vv:12:4: error: cannot assign to field `error`: expected `?string`, not `IError`
   10 |     some_struct := result_function() or {
   11 |         Struct{
   12 |             error: err
      |             ~~~~~~~~~~
   13 |         }
   14 |     }
